### ISSUE TITLE
TextBox error message not disappearing on correct input

### ISSUE THREAD

**Issue by KDouble48** [AvaloniaUI/Avalonia#16229]
### Describe the bug

The TextBox error message persists after re-entering previous value into TextBox.

### To Reproduce

Add a Property in ViewModel: `public int Number { get; set; }`
Add a TextBox: `<TextBox Text="{Binding Number}"
             TextInputOptions.ContentType="Number"
             Watermark="Enter a number" />`
Remove default "0" in TextBox -> get expected `InvalidCastException` error message
Input "0" again into TextBox -> error message persists

Works with any number (also floats) if the first valid input equals the previous valid input.
Error message disappears after another valid input was given.

### Expected behavior

The error message should disappear after any valid input.

### Avalonia version

11.1.0-rc2

### OS

Windows

**Comment by maxkatz6** [comment]
@KDouble48 was it reproducible in 11.0 versions?

**Comment by KDouble48** [comment]
No, it was working as expected. RC1 had a problem of crashing the app.

**Comment by IanRawley** [comment]
Sounds like #16150 might fix this, assuming the issue is that the binding aren't reacting to the new input because it's the same as the last input they recognized.

**Comment by maxkatz6** [comment]
@IanRawley that PR should be already included in rc2 though.

**Comment by MrJul** [comment]
With #19477 the issue is now solved.
